### PR TITLE
Unlimited invites for Zerocracy QA

### PR DESCRIPTION
**Pull request by g4s8**
#1410 - remove invite limitations for zerocracy QA users

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1413](https://github.com/zerocracy/farm/issues/1413) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1413](https://github.com/zerocracy/farm/issues/1413) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1921762924); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy [hunk 1]
```diff
@@ -59,8 +63,28 @@ def exec(Project pmo, XML xml) {
...
     )
   }
+  // we allow to invite new students without limitations to PMO users
+  // and 'farm' (C3NDPUA8L) QA users
+  // see https://github.com/zerocracy/farm/issues/1410
+  Scalar<Boolean> force = new Or(
+    new ListOf<>(
+      new Scalar<Boolean>() {
+        @Override
+        Boolean value() throws Exception {
```
Comment (krzyk) [inline src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy]: @g4s8 is explicit scalr here required? wouldn't
```
{ -> new Roles(new Pmo(farm)).bootstrap().hasAnyRole(author)}
```
also work in groovy?

File: src/test/java/com/zerocracy/pmo/PeopleTest.java [hunk 2]
```diff
@@ -322,6 +327,19 @@ public void inviteSixteen() throws Exception {
...
 
+    @Test
+    public void inviteForce() throws Exception {
+        final String mentor = "supermentor";
+        final People people = new People(new FkFarm()).bootstrap();
+        new And(
+            (String std) -> people.invite(std, mentor, true),
+            new Mapped<>(
+                (Integer num) -> String.format("std%d", num),
+                new RangeOf<>(0, Tv.TWENTY, x -> x + 1)
```
[the remark this answers is not in the document]
Comment (g4s8) [reply src/test/java/com/zerocracy/pmo/PeopleTest.java]: @krzyk right, but we can't use policy values in test, so I replaced it with 16 (second parameter in `new Policy().get("1.max-students", 16);`)

File: src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy (hunk 1 above)
Comment (g4s8) [reply src/main/groovy/com/zerocracy/stk/pmo/profile/invite_friend.groovy]: @krzyk no, groovy can't convert `{...}` to `Scalar` here

**Comment by g4s8** [comment]
@krzyk fixed, please look again

**Comment by krzyk** [comment]
@g4s8 looks good

**Comment by krzyk** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1413](https://github.com/zerocracy/farm/issues/1413) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +31 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**